File: backend/src/types/spot.ts
export interface Spot {
  id: string;
  name: string;
  lat: number;
  lng: number;
  source: 'osm' | 'ugc' | 'opentrip' | 'flickr' | 'mix';
  categories?: string[];
  score: number;
  photo_url?: string;
  description?: string;
  last_enriched_at?: string;
  created_at: string;
  updated_at: string;
}

export interface SpotStats {
  spot_id: string;
  photo_density: number;
  recency_trend: number;
  opentrip_popularity: number;
  flickr_photo_count: number;
  updated_at: string;
}

export interface SpotWithDistance extends Spot {
  distance_m?: number;
}

export interface CreateSpotInput {
  name: string;
  lat: number;
  lng: number;
  source: Spot['source'];
  categories?: string[];
  description?: string;
}

File: Photospots(Test)/Photospots(Test)/Models/Spot.swift
import Foundation
import CoreLocation

struct Spot: Identifiable, Codable {
    let id: String
    let name: String
    let lat: Double
    let lng: Double
    let source: String // Using String for simplicity, can be enum later
    let categories: [String]?
    let score: Double
    let photoUrl: String?
    let description: String?
    let lastEnrichedAt: String?
    let createdAt: String
    let updatedAt: String
    
    enum CodingKeys: String, CodingKey {
        case id, name, lat, lng, source, categories, score
        case photoUrl = "photo_url"
        case description
        case lastEnrichedAt = "last_enriched_at"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
    
    var coordinate: CLLocationCoordinate2D {
        CLLocationCoordinate2D(latitude: lat, longitude: lng)
    }
}
